### ISSUE TITLE
text nodes may or may not be incorrectly parsed

### ISSUE THREAD

**Issue by Pomax** [mozilla/slowparse#71]
the DOM that is build for text nodes such as 

```
<ul>
    <li>Hello Everyone
</ul>
```

ends up treating the text node as a sibling to the `<li>` rather than as a child. Depending on whether slowparsing is doing the right thing this may or may not require fixing

**Comment by lemiant** [comment]
I found an example where improper behavior for optional closing tags creates a parsing error, the following will raise a MISMATCHED_CLOSE_TAG error in slowparse:

```
<ol>
    <li>Hello <a>World</a>
</ol>
```

However this is actually valid HTML and passes the HTML5 validator
http://validator.w3.org/check?uri=http%3A%2F%2Fhost.codingproject.net%2Fslowparse_error_example.html

**Comment by Pomax** [comment]
you're right. the spec says `An li element's end tag may be omitted if the li element is immediately followed by another li element or if there is no more content in the parent element.` so this is definitely a bug.
